storage/gcp/gcp_test.go

```diff
@@ -304,9 +305,72 @@ func TestBundleRoundtrip(t *testing.T) {
 	}
 }
 
+func TestPublishCheckpoint(t *testing.T) {
+	ctx := context.Background()
+
+	close := newSpannerDB(t)
+	defer close()
+
+	s, err := newSpannerSequencer(ctx, "projects/p/instances/i/databases/d", 1000)
+	if err != nil {
+		t.Fatalf("newSpannerSequencer: %v", err)
+	}
+
+	for _, test := range []struct {
+		name            string
+		cpModifiedAt    time.Time
+		publishInterval time.Duration
+		wantUpdate      bool
+	}{
+		{
+			name:            "works ok",
+			cpModifiedAt:    time.Now().Add(-15 * time.Second),
+			publishInterval: 10 * time.Second,
+			wantUpdate:      true,
+		}, {
+			name:            "too soon, skip update",
+			cpModifiedAt:    time.Now().Add(-5 * time.Second),
+			publishInterval: 10 * time.Second,
+			wantUpdate:      false,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			m := newMemObjStore()
+			storage := &Storage{
+				objStore:    m,
+				sequencer:   s,
+				entriesPath: layout.EntriesPath,
+				newCP:       func(size uint64, hash []byte) ([]byte, error) { return []byte(fmt.Sprintf("%d/%x,", size, hash)), nil },
+			}
+			// Call init so we've got a zero-sized checkpoint to work with.
+			if err := storage.init(ctx); err != nil {
+				t.Fatalf("storage.init: %v", err)
+			}
+			m.delete(layout.CheckpointPath)
```

Comment:
It looks like you're deleting the checkpoint here so that you can tell that something has been written a few lines down. This seems a bit hacky because the checkpoint would never be missing in reality. Is there a better way, or are you happy with this?